org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/util/IResourcesSetupUtil.java

```diff
@@ -425,4 +456,13 @@ public static String printMarker(IMarker[] markers) throws CoreException {
 		}
 		return result.toString();
 	}
+
+	public static void waitForJdtIndex() {
```

Comment:
Could it happen that this waits indefinitely?

Reply:
No, as far as I understand. Unless there's a bug in jdt. Similar to build(): if a builder is buggy and loops the whole build gets stuck. At least, that's my understanding.